File: Python /9_palindrome_number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x=str(x)
        l=0
        r=len(x)-1
        while l<r:
            if x[l] != x[r]:
                return False
            l+=1
            r-=1
        return True
        
opt = Solution()
print(opt.isPalindrome(121))

File: C++/Leetcode/9.palindrome_number.cpp
class Solution {
public:
    bool isPalindrome(int number) {
        if (number < 0 || !number  && number%10  ) return false;
        long long int reverse_number = 0;
        long long int temp_number = number;
        while (temp_number){
            reverse_number = reverse_number*10 + temp_number%10;
            temp_number /= 10;
        }
        
        return reverse_number == number ? true:false;
        
      


    }
};
